File: CS/InventoryChart_2/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace InventoryChart
{
    public partial class Form1 : Form
    {
        NorthwindEntities db = new NorthwindEntities();

        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            var categorySales = (from c in db.Categories
                                 select new
                                 {
                                     CategoryID = c.Category_ID,
                                     CategoryName = c.Category_Name,
                                     CategorySum = c.Products.Sum(p => p.Units_In_Stock)
                                 }
                     ).ToList();
            this.inventoryChart.Series["CategoryUnits"].DataSource = categorySales;
            this.inventoryChart.Series["CategoryUnits"].ArgumentDataMember = "CategoryName";
            this.inventoryChart.Series["CategoryUnits"].ValueDataMembers.AddRange(new string[] { "CategorySum" });

        }

    }
}

File: VB/InventoryChart_2/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace InventoryChart
	Partial Public Class Form1
		Inherits Form
		Private db As New NorthwindEntities()

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim categorySales = (From c In db.Categories _
			                     Select New With {Key .CategoryID = c.Category_ID, Key .CategoryName = c.Category_Name, Key .CategorySum = c.Products.Sum(Function(p) p.Units_In_Stock)}).ToList()
			Me.inventoryChart.Series("CategoryUnits").DataSource = categorySales
			Me.inventoryChart.Series("CategoryUnits").ArgumentDataMember = "CategoryName"
			Me.inventoryChart.Series("CategoryUnits").ValueDataMembers.AddRange(New String() { "CategorySum" })

		End Sub

	End Class
End Namespace
